docker-plugin/src/main/resources/com/nirima/jenkins/plugins/docker/DockerTemplate/config.groovy

```diff
@@ -0,0 +1,80 @@
+package com.nirima.jenkins.plugins.docker.DockerTemplate
+
+import com.nirima.jenkins.plugins.docker.DockerTemplate
+import com.nirima.jenkins.plugins.docker.utils.Functions
+
+def f = namespace(lib.FormTagLib)
+def st = namespace("jelly:stapler")
+
+// when added to heteroList
+if (instance == null) {
+    instance = new DockerTemplate();
+}
+
+f.property(field: "dockerTemplateBase")
+
+f.entry(title: _("Instance Capacity"), field: "instanceCapStr") {
+    f.textbox()
+}
+
+f.entry(title: _("Remote Filing System Root"), field: "remoteFs") {
+    f.textbox()
+}
+
+f.entry(title: _("Labels"), field: "labelString",
+        help: "/descriptor/com.nirima.jenkins.plugins.docker.DockerSlave/help/labelString") {
+    f.textbox()
+}
+
+f.slave_mode(name: "mode", node: instance)
+
+f.advanced(title: _("Experimental Options"), align: "left") {
+    f.dropdownList(name: "retentionStrategy", title: _("Availability"),
+            help: "/help/system-config/master-slave/availability.html") {
+        Functions.dockerRetentionStrategyDescriptors.each { sd ->
+            if (sd != null) {
+                def prefix = sd.displayName.equals("Docker Once Retention Strategy") ? "" : "Experimental: "
+
+                f.dropdownListBlock(value: sd.clazz.name, name: sd.displayName,
+                        selected: instance.retentionStrategy != null ?
+                                instance.retentionStrategy.descriptor.equals(sd) : false,
+                        title: prefix + sd.displayName) {
+                    descriptor = sd
+                    f.invisibleEntry() {
+                        input(type: "hidden", name: "stapler-class", value: sd.clazz.name)
+                    }
+                    st.include(from: sd, page: sd.configPage, optional: "true")
+                }
+            }
+        }
+    }
+
+    f.entry(title: _("# of executors"), field: "numExecutors") {
+        f.number(default: "1")
+    }
+}
+
+f.dropdownList(name: "launcher", title: _("Launch method"),
+        help: descriptor.getHelpFile('launcher')) {
+    Functions.dockerComputerLauncherDescriptors.each { ld ->
+        if (ld != null) {
```

Comment:
I just converted crazy jelly, hope something can be simplified in future, now i don't want break something.